### PR TITLE
[FIX] hr_expense: Update payment states after moves updated

### PR DESCRIPTION
**Pull request by nboulif**
Issue

	- Configure bank journal to post at bank reconciliation
	- Create 2 expenses paid by company
	- Create a report for the two expenses
	  (make sure bank journal configured in the first step is used)
	- Submit, approve and post journal entries

	Error message : "You cannot modify a journal entry linked to
	a posted payment."

Cause

	Since the payment state of expenses lines is updated one by one
	in related move, the first expense had already a 'posted'
	state when updating second expense line, who go against
	_validate_move_modification validation.

Solution

	Update payments state only after updating all moves lines.

opw-2275686

### CODE REVIEW

**Comment by nboulif** [comment]
Ref https://github.com/odoo/odoo/pull/51650

File: addons/hr_expense/models/hr_expense.py [hunk 1]
```diff
@@ -455,6 +455,8 @@ def action_move_create(self):
 
         move_to_keep_draft = self.env['account.move']
 
+        payments = []
```
Comment (fmdl) [inline addons/hr_expense/models/hr_expense.py:458]: `company_payments = self.env['account.payment']`

File: addons/hr_expense/models/hr_expense.py [hunk 2]
```diff
@@ -496,13 +498,16 @@ def action_move_create(self):
 
             if expense.payment_mode == 'company_account':
                 if journal.post_at == 'pay_val':
-                    payment.state = 'reconciled'
+                    payments.append((payment, 'reconciled'))
```
Comment (fmdl) [inline addons/hr_expense/models/hr_expense.py:501]:
```python
if expense.payment_mode == 'company_account':
     company_payments |= payment
     if journal.post_at == 'bank_rec':
           move_to_keep_draft |= move
```

File: addons/hr_expense/models/hr_expense.py [hunk 3]
```diff
@@ -496,13 +498,16 @@ def action_move_create(self):
...
                 elif journal.post_at == 'bank_rec':
-                    payment.state = 'posted'
+                    payments.append((payment, 'posted'))
                     move_to_keep_draft |= move
 
                 expense.sheet_id.paid_expense_sheets()
 
+        for payment, state in payments:
+            payment.state = state
+
```
Comment (fmdl) [inline addons/hr_expense/models/hr_expense.py:510]:
```
company_payments.filtered(lambda x: x.journal_id.post_at == 'pay_val').write({'state':'reconciled'})
company_payments.filtered(lambda x: x.journal_id.post_at == 'bank_rec').write({'state':'posted'})
```

**Comment by nboulif** [comment]
@fmdl Lot more cleaner, thanks for the review. Done :+1:

**Comment by simongoffin** [comment]
robodoo r+

**Pull request closed without merging**